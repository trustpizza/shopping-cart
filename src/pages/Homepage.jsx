import { useState } from "react";
import { Link } from "react-router-dom";

const Homepage = () => {

  return (
    <>
    <div>
      <h1>Bananazon!</h1>
    </div>
    </>
  )
}

export default Homepage;