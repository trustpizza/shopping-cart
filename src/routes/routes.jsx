import Homepage from "../pages/Homepage";
import Shop from "../pages/Shop";

import Error404 from "../pages/error/404";

const routes = [
  {
    path: "/",
    element: <Homepage />,
    errorElement: <Error404 />
  },
  {
    path: "shop",
    element: <Shop />
  }
]

export default routes;