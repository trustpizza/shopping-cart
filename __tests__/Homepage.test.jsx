import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom"; // Import MemoryRouter
import routes from "../src/routes/routes";

import Homepage from "../src/pages/Homepage";

describe("Homepage Component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/']}>  {/* Simulate the initial route */}
        <Homepage />
      </MemoryRouter>
    );

    // Take a snapshot of the rendered component
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("renders all images with correct alt attributes", () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <Homepage />
      </MemoryRouter>
    );

    // Define expected images and their corresponding alt texts
    const expectedImages = [
      { alt: "Bananazon!", src: "src/assets/logo-black.png" },
      { alt: "Bananazon!", src: "src/assets/logo-white.png" },
    ];

    const images = screen.getAllByRole("img");

    expect(images.length).toBe(expectedImages.length);

    images.forEach((img, index) => {
      expect(img).toHaveAttribute("alt", expectedImages[index].alt);
      expect(img).toHaveAttribute("src", expect.stringContaining(expectedImages[index].src));
    });
  });
})
