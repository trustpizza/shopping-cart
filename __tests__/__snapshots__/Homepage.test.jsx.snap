// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage Component > matches the snapshot 1`] = `
<DocumentFragment>
    
  <div>
    <h1>
      Bananazon!
    </h1>
  </div>
</DocumentFragment>
`;
